package.json

```diff
@@ -31,14 +31,15 @@
     ]
   },
   "scripts": {
+    "postinstall": "npm run compile",
```

Comment:
You can use the `prepare` hook here and it will only run if dev dependencies are installed. As compile relies on these dependencies this will prevent errors installing only for production. 

(It would be good to double check this works as expected before committing to it)

Reply:
Ok moved it to prepare also did `rm -rf node_modules && npm i` and confirm it works